src/views/List.jsx

```diff
@@ -1,15 +1,69 @@
+import { useState, useEffect } from 'react';
 import { ListItem } from '../components';
 
 export function List({ data }) {
+	// state for input
+	const [input, setInput] = useState('');
+	const [searchData, setSearchData] = useState(data);
+	const [isValid, setIsValid] = useState(false);
+	const [searchLength, setSearchLength] = useState(1);
+	// Handle event
+	const handleInputChange = (e) => {
+		setInput(e.target.value);
+	};
+
+	const handleInputClear = () => {
+		setInput('');
+	};
+
+	useEffect(() => {
+		const searchRegex = new RegExp(input, 'i');
+		const filteredData = data.filter((item) => searchRegex.test(item.name));
+		setSearchData(filteredData);
+	}, [data, input]);
```

Comment:
If you use an array or an object in `useEffect`'s dependency array then the `useEffect` will trigger every single render because an object or an array are never the same as themselves. Consider changhing `data` to `data.length`.

Reply:
I made this change, but the terminal returned a warning saying "React Hook has missing dependency 'data' ". It returned no error in the other instance you recommended changing. Which is preferred?